src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -1352,7 +1354,12 @@ private DynamicObject finishPack(int formatLength, BytesResult result) {
                 makeLeafRopeNode = insert(RopeNodes.MakeLeafRopeNode.create());
             }
 
-            final DynamicObject string = createString(makeLeafRopeNode.executeMake(
+            if (makeStringNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                makeStringNode = insert(StringNodes.MakeStringNode.create());
+            }
+
+            final DynamicObject string = makeStringNode.fromRope(makeLeafRopeNode.executeMake(
```

Comment:
Style: this might be nice to use the pattern: https://github.com/graalvm/truffleruby/blob/master/doc/contributor/code-patterns.md#the-lazy-pattern

Reply:
The initialization above should be extracted to a method according to our code patterns (so the main logic is clear and uncluttered by these kind of details).